waspc/data/Generator/templates/sdk/wasp/client/operations/queries/core.ts

```diff
@@ -18,7 +18,7 @@ import {
 // Details here: https://github.com/wasp-lang/wasp/issues/2017
 export function makeQueryCacheKey<Input, Output>(
   query: Query<Input, Output>,
-  payload: Input
+  payload?: Input
```

Comment:
`payload !== undefined` hints at this being optional

Reply:
I'm not sure this is equivalent to what we had. Perhaps `Input` can be undefined and we only want this check to fire then (i.e., we never call this function without a payload).

Can you double check?

Reply:
Reverted this change and use `as Input` assertion in the calling place.